src/components/ToggleSwitch.js

```diff
@@ -93,73 +63,103 @@ const ToggleSwitch = React.createClass({
         fontSize: '12px',
         position: 'relative'
       },
-      inactiveLabel: {
-        color: this.props.inactiveColor
+      icon: {
+        fill: StyleConstants.Colors.WHITE,
+        position: 'absolute',
+        top: '0px',
+        zIndex: 2
+      },
+      trueIcon: {
+        left: '0px'
+      },
+      falseIcon: {
+        right: '0px'
       },
       label: {
         cursor: 'pointer',
         fontWeight: 'bold'
       },
-      left: {
-        left: this.props.trackStyle.padding || '2px',
-        transition: 'all .1s'
+      inactiveLabel: {
+        color: StyleConstants.Colors.FOG
       },
-      right: {
-        right: this.props.trackStyle.padding || '2px',
-        transition: 'all .1s'
+      activeLabel: {
+        color: StyleConstants.Colors.PRIMARY
       },
       toggle: {
         backgroundColor: StyleConstants.Colors.WHITE,
         borderRadius: '100%',
-        height: this.props.toggleStyle.height || '20px',
+        height: '20px',
         position: 'absolute',
-        width: this.props.toggleStyle.width || '20px'
+        width: '20px',
+        transition: 'all 0.5s ease',
+        zIndex: 3
+      },
+      falseToggle: {
+        left: '2px'
+      },
+      trueToggle: {
+        left: '20px'
       },
       track: {
-        backgroundColor: StyleConstants.Colors.FOG,
-        borderRadius: this.props.trackStyle.height || '20px',
+        borderRadius: '20px',
         cursor: 'pointer',
         display: 'inline-block',
-        height: this.props.trackStyle.height || '20px',
+        height: '20px',
         margin: '0 10px',
-        padding: this.props.trackStyle.padding || '2px',
+        padding: '2px',
         position: 'relative',
+        transition: 'all 0.5s ease',
         verticalAlign: 'middle',
-        width: this.props.trackStyle.width || '38px'
+        width: '38px',
+        zIndex: 1
+      },
+      trueTrack: {
+        backgroundColor: StyleConstants.Colors.CHARCOAL
+      },
+      falseTrack: {
+        backgroundColor: StyleConstants.Colors.ASH
       }
-    };
+    }, this.props.styles);
 
     return (
-      <div className='toggle-switch-component' style={[styles.component, this.props.componentStyle]}>
+      <div className='toggle-switch-component' style={styles.component}>
         {this._renderLeftLabel(styles)}
-        <div className='toggle-switch-track' onClick={this._handleToggle} style={[styles.track, this.props.trackStyle]} >
-          <div className='toggle-switch-toggle' style={[styles.toggle, styles[this.state.activePosition], this.props.toggleStyle]}></div>
-          {this._getChildren()}
+        <div
+          className='toggle-switch-track'
+          onClick={this._handleToggle}
+          style={Object.assign({}, styles.track, styles[this.state.checked + 'Track'])}
+        >
+          <div className='toggle-switch-toggle' style={Object.assign({}, styles.toggle, styles[this.state.checked + 'Toggle'])}></div>
```

Comment:
This just allows someone to style the elements in a normal css sheet instead of the style prop. Some further work needs to be done to add animation support with dynamic class names. It would easily be accomplished with a library such as https://github.com/JedWatson/classnames